metadata_backend/api/middlewares.py

```diff
@@ -0,0 +1,50 @@
+"""Middleware methods for server."""
+import json
+from typing import Callable
+
+from aiohttp.web import HTTPError, Request, Response, middleware
+
+
+def error_middleware() -> Callable:
+    """Middleware for handling exceptions recieved from the API methods.
+
+    :returns: Handled response
+    """
+
+    @middleware
+    async def http_error_handler(req: Request, handler: Callable) -> Response:
+        """Handle HTTP errors.
+
+        :param req: A request instance
+        :param handler: A request handler
+        :returns: JSON response for the error
+        """
+        try:
+            response = await handler(req)
+            if response.status == 404:
```

Comment:
I don't think this is needed separately here, since 404's are also catched with except-clause below.

You can just `return await handler(req)`

Reply:
Or we could actually add forward to frontend, where react renders 404 page?